### PR TITLE
ChatEntry notebook improvements

### PR DESCRIPTION
**Pull request by MarcSkovMadsen**
FYI @ahuang11

### CODE REVIEW

**Comment by MarcSkovMadsen** [comment]
I've change the `value` parameter type back to `Parameter` such that the `ChatEntry` can display any object that panel can display. Not just a subset.

```python
import panel as pn
from panel.widgets import ChatEntry
import numpy as np
from asyncio import sleep

def mpl_figure():
    import matplotlib.pyplot as plt
    fig = plt.figure()
    ax = fig.add_subplot(111)
    ax.plot(np.random.rand(10, 2))
    plt.close(fig)
    return fig

async def async_func():
    return "hello"

async def async_generator():
    await sleep(1)
    yield "hello"
    await sleep(2)
    yield "hello world"

pn.Column(
    ChatEntry(value=mpl_figure()),
    ChatEntry(value=async_func),
    ChatEntry(value=async_generator),
).servable()
```

File: examples/reference/widgets/ChatEntry.ipynb [hunk 1]
```diff
@@ -59,14 +59,16 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "pn.widgets.ChatEntry(value=\"Good to see you!\", user=\"Happy User\", avatar=\"😀\")"
+    "from panel.widgets import ChatEntry\n",
```
Comment (MarcSkovMadsen) [inline examples/reference/widgets/ChatEntry.ipynb:62]: I import `ChatEntry` to make the code shorter and easier to read.

File: examples/reference/widgets/ChatEntry.ipynb [hunk 2]
```diff
@@ -75,8 +77,73 @@
...
+    "dataframe = pd.DataFrame(data, columns=list(\"AB\"))\n",
+    "\n",
+    "fig = plt.figure(figsize=(5,2.5))\n",
+    "ax = fig.add_subplot(111)\n",
+    "ax.plot(data)\n",
+    "plt.close(fig)\n",
+    "\n",
+    "pn.Column(\n",
+    "    ChatEntry(\n",
+    "        value=pn.widgets.Button(name=\"CLICK ME\", button_type=\"primary\"),\n",
```
Comment (MarcSkovMadsen) [inline examples/reference/widgets/ChatEntry.ipynb:94]: This is the example where the `Button` displays as `False`?

File: panel/widgets/chat.py [hunk 3]
```diff
@@ -204,8 +204,8 @@ class ChatEntry(CompositeWidget):
         Whether to display the timestamp of the message.
     """
 
-    value = param.ClassSelector(class_=(Viewable, str, int, float, _FileInputMessage), doc="""
-        The message contents. Can be a string, pane, widget, layout, etc.""")
+    value = param.Parameter(doc="""
```
Comment (MarcSkovMadsen) [inline panel/widgets/chat.py:207]: I want to be able to display any object that Panel can display. Not only a limited subset 😄

File: examples/reference/widgets/ChatEntry.ipynb (hunk 2 above)
Comment (ahuang11) [reply examples/reference/widgets/ChatEntry.ipynb:94]: I changed it from Parameter because of this; Parameter casts widgets into Str

File: panel/widgets/chat.py (hunk 3 above)
Comment (ahuang11) [reply panel/widgets/chat.py:207]: If its Parameter most widgets input are broken that is why I changed it to limited subset

File: examples/reference/widgets/ChatEntry.ipynb (hunk 2 above)
Comment (MarcSkovMadsen) [reply examples/reference/widgets/ChatEntry.ipynb:94]: Is that a bug @philippjfr ?

File: panel/widgets/chat.py (hunk 3 above)
Comment (MarcSkovMadsen) [reply panel/widgets/chat.py:207]: I will try a workaround now by wrapping it in a bound function.

Comment (MarcSkovMadsen) [reply panel/widgets/chat.py:207]: I can see its already wrapped. The below is a minimum reproducible, test example 

```python
import panel as pn
import param

class Entry(pn.widgets.base.Reactive):
    value = param.Parameter()

entry = Entry(
    value=pn.widgets.Button(name="CLICK ME", button_type="primary"),
)
assert isinstance(entry.value, pn.widgets.Button)
```

Would you consider this a bug @philippjfr ?

Comment (MarcSkovMadsen) [reply panel/widgets/chat.py:207]: When debugging, I can see that there is an attribute `_ignored_refs` that can be set to `('value',)`. I believe that will solve the problem.

Comment (MarcSkovMadsen) [reply panel/widgets/chat.py:207]: It works. I'm closing this one.

File: examples/reference/widgets/ChatEntry.ipynb (hunk 2 above)
Comment (MarcSkovMadsen) [reply examples/reference/widgets/ChatEntry.ipynb:94]: I've solved this by setting `ChatEntry._ignored_refs=('value',)`. Closing this one.

File: panel/pane/base.py [hunk 4]
```diff
@@ -545,7 +545,7 @@ class ReplacementPane(PaneBase):
 
     _pane = param.ClassSelector(class_=Viewable)
 
-    _ignored_refs: ClassVar[Tuple[str]] = ['object']
+    _ignored_refs: ClassVar[Tuple[str,...]] = ('object',)
```
Comment (MarcSkovMadsen) [inline panel/pane/base.py:548]: Please note the types and values of `_ignore_refs` were incorrect. I believe I've fixed them correctly in the places where needed @philippjfr

**Comment by MarcSkovMadsen** [comment]
I've fixed the issue with the `Button`s being turned to into `str`s.

File: panel/widgets/chat.py [hunk 5]
```diff
@@ -203,9 +203,10 @@ class ChatEntry(CompositeWidget):
     show_timestamp : bool
         Whether to display the timestamp of the message.
     """
+    _ignored_refs: ClassVar[Tuple[str,...]] = ('value',)
```
Comment (MarcSkovMadsen) [inline panel/widgets/chat.py:206]: This solved the issue with `Button`s and other widgets being turned into `str`s @ahuang11 and @philippjfr .

I believe its the right solution. But I'm not familiar with this `attribute` and why its there?

File: panel/pipeline.py [hunk 6]
```diff
@@ -165,7 +165,7 @@ class Pipeline(Viewer):
 
     previous = param.Action(default=lambda x: x.param.trigger('previous'))
 
-    _ignored_refs: ClassVar[Tuple[str]] = ('next_parameter', 'ready_parameter')
+    _ignored_refs: ClassVar[Tuple[str,...]] = ('next_parameter', 'ready_parameter')
```
Comment (ahuang11) [inline panel/pipeline.py:168]:
```suggestion
    _ignored_refs: ClassVar[Tuple[str, ...]] = ('next_parameter', 'ready_parameter')
```

File: panel/pane/holoviews.py [hunk 7]
```diff
@@ -678,7 +678,7 @@ class Interactive(PaneBase):
 
     priority: ClassVar[float | bool | None] = None
 
-    _ignored_refs: ClassVar[Tuple[str]] = ['object']
+    _ignored_refs: ClassVar[Tuple[str,...]] = ('object',)
```
Comment (ahuang11) [inline panel/pane/holoviews.py:681]:
```suggestion
    _ignored_refs: ClassVar[Tuple[str, ...]] = ('object',)
```

File: examples/reference/widgets/ChatEntry.ipynb [hunk 8]
```diff
@@ -75,8 +77,76 @@
...
+    "import pandas as pd\n",
+    "import numpy as np\n",
+    "\n",
+    "data = np.random.rand(5, 2)\n",
+    "dataframe = pd.DataFrame(data, columns=list(\"AB\"))\n",
+    "\n",
+    "fig = plt.figure(figsize=(5,2.5))\n",
+    "ax = fig.add_subplot(111)\n",
+    "ax.plot(data)\n",
+    "plt.close(fig)\n",
```
Comment (ahuang11) [inline examples/reference/widgets/ChatEntry.ipynb:90]: I prefer less unrelated code for reference gallery; perhaps something like (haven't tested):
```python
import matplotlib.pyplot as plt

figs, axes = plt.subplots(1, 1)
axes[0].plot([1, 2, 3])
```
would suffice to demo support for Matplotlib?

**Review by ahuang11 — CHANGES_REQUESTED** [review]
Thanks for expanding it; some thoughts.

**Review by ahuang11 — APPROVED** [review]
Going to merge this now and fix the pn.bind in the main branch!
